### PR TITLE
Fix(Issue no. 1330) for Relaxed Mocking Value When Property is Nested Value Class

### PR DESCRIPTION
**Pull request by kpadhiamex**
Fix for the Issue: https://github.com/mockk/mockk/issues/1330 

Fixed for given nested value classes when relaxed mocking is enabled then value successfully verified.
Added couple of test(s) so that it will be verified in every PR.

**Root Cause**
Because of this PR's: https://github.com/mockk/mockk/pull/1314/files change in the below code in the else part, this issue was broken. Instead of returning this in the else block which returns the value class, instead check if there is any boxedValue and return the same if exists else return the same value class.

**Earlier**
```
return if (resultType == expectedReturnType && !(isReturnNullable && isPrimitive)) {
    this.boxedValue
} else {
    this
}
```

**Now**
```
return if (resultType == expectedReturnType && !(isReturnNullable && isPrimitive)) {
    this.boxedValue
 } else {
    this.boxedValue ?: this
 }
```

### CODE REVIEW

**Comment by kpadhiamex** [comment]
@Raibaz Can I ask, kindly review this PR if you are available?

**Comment by Raibaz** [comment]
Hi @kpadhiamex, thanks for looking into this!

PR looks good to me, but to merge it I need the builds to pass. Can you please rebase on top of `master`? I just pushed a fix to the build process.

**Review by Raibaz — APPROVED** [review]

**Comment by kpadhiamex** [comment]
Hi @Raibaz , Many thanks for your comments and approval. I just merged master branch, could you please have a look and approve the workflow which is waiting for approval?

`1 workflow awaiting approval`

**Pull request merged**
